internal/operators/ocs/ocs_operator.go

```diff
@@ -82,31 +85,84 @@ func (o *ocsOperator) GenerateManifests(cluster *common.Cluster) (*api.Manifests
 }
 
 // GetCPURequirementForWorker provides worker CPU requirements for the operator
-func (o *ocsOperator) GetCPURequirementForWorker(context.Context, *common.Cluster) (int64, error) {
-	return 0, nil
+func (o *ocsOperator) GetCPURequirementForWorker(cluster *common.Cluster) (int64, error) {
+	return o.getPerHostCPURequirement(cluster)
 }
 
 // GetCPURequirementForMaster provides master CPU requirements for the operator
-func (o *ocsOperator) GetCPURequirementForMaster(context.Context, *common.Cluster) (int64, error) {
-	return 0, nil
+func (o *ocsOperator) GetCPURequirementForMaster(cluster *common.Cluster) (int64, error) {
+	return o.getPerHostCPURequirement(cluster)
 }
 
 // GetMemoryRequirementForWorker provides worker memory requirements for the operator in MB
-func (o *ocsOperator) GetMemoryRequirementForWorker(context.Context, *common.Cluster) (int64, error) {
-	return 0, nil
+func (o *ocsOperator) GetMemoryRequirementForWorker(cluster *common.Cluster) (int64, error) {
+	return o.getPerHostMemoryRequirement(cluster)
 }
 
 // GetMemoryRequirementForMaster provides master memory requirements for the operator
-func (o *ocsOperator) GetMemoryRequirementForMaster(context.Context, *common.Cluster) (int64, error) {
-	return 0, nil
+func (o *ocsOperator) GetMemoryRequirementForMaster(cluster *common.Cluster) (int64, error) {
+	return o.getPerHostMemoryRequirement(cluster)
 }
 
-// GetDisksRequirementForMaster provides a number of disks required in a master
-func (o *ocsOperator) GetDisksRequirementForMaster(context.Context, *common.Cluster) (int64, error) {
-	return 0, nil
+// getHostMemoryRequirement returns the minimum amount of memory in Bytes the host must have to install the OCS operator
+func (o *ocsOperator) getPerHostMemoryRequirement(cluster *common.Cluster) (int64, error) {
+	depType, err := o.ocsValidator.IdentifyDeploymentTypeForResource(&cluster.Cluster, memory)
+	if err != nil {
+		return 0, err
+	}
+	var mem, count int64
+
+	switch depType {
+	case compact:
+		mem = o.ocsValidatorConfig.OCSRequiredCompactModeRAMGB * int64(units.GB)
+		count = 3 // in compact mode we deploy only with 3 masters
+	case minimal, standard:
+		mem = o.ocsValidatorConfig.OCSMinimumRAMGB * int64(units.GB)
+		count = o.getWorkerHostsCount(cluster) // OCS only deploys only in workers when not in compact mode.
+	default:
+		return 0, fmt.Errorf("invalid OCS deployment type %v", depType)
+	}
+	if count == 0 {
+		return 0, fmt.Errorf("unable to find hosts with valid inventory")
+	}
+	return mem / count, nil
+}
+
+// getHostCPURequirement returns the minimum amount of CPU core count the cluster must have to install the OCS operator
+func (o *ocsOperator) getPerHostCPURequirement(cluster *common.Cluster) (int64, error) {
```

Comment:
The per host requirements are different and yet to be implemented. As we only perform aggregate resource validations,
for eg, if in compact mode (3 masters), its not compulsary that per host must have atleast 12 cpu, we rely on aggregate 36 cpu and can be present as 7,15,14.

Reply:
What do you suggest at this point in time? We are implementing per host validation to ensure that all operators can be deployed based on the available hardware.